spec/queries/json_api_spec.rb

```diff
@@ -1,33 +1,58 @@
 require 'spec_helper'
+require 'faraday_middleware'
 require 'app/queries/json_api'
+require 'active_support/core_ext/object/blank'
 
 RSpec.describe JsonApi do
   describe '#get' do
-    it 'does not raise an error' do
-      json_api = JsonApi.new
-
-      expect { json_api.get('asfasf') }.not_to raise_error
-    end
-
     context 'with a path matching properties?postcode=' do
-      it 'returns a valid response' do
-        json_api = JsonApi.new
+      it 'parses a JSON response' do
+        json_api = JsonApi.new(api_root: 'http://hackney.api:8000')
+        stub_request(:get, 'http://hackney.api:8000/properties?postcode=A1%201AA')
+          .to_return(
+            body: [{ 'property_reference' => 'abc123', 'short_address' => '1 some road' }].to_json,
+            headers: { content_type: 'application/json' }
+          )
 
         result = json_api.get('properties?postcode=A1 1AA')
 
-        expect(result).to be_an(Array)
-        expect(result.first.keys).to include('property_reference', 'short_address')
+        expect(result).to eq [{ 'property_reference' => 'abc123', 'short_address' => '1 some road' }]
       end
     end
 
     context 'with a path matching properties/:property_reference' do
-      it 'returns a valid response' do
-        json_api = JsonApi.new
+      it 'parses a JSON response' do
+        json_api = JsonApi.new(api_root: 'http://hackney.api:8000')
+        stub_request(:get, 'http://hackney.api:8000/properties/zxc098')
+          .to_return(
+            body: { 'uprn' => 'xyz987', 'property_reference' => 'abc123', 'short_address' => '1 some road' }.to_json,
+            headers: { content_type: 'application/json' }
+          )
 
         result = json_api.get('properties/zxc098')
 
-        expect(result).to be_a(Hash)
-        expect(result.keys).to include('property_reference', 'short_address', 'uprn')
+        expect(result).to eq('uprn' => 'xyz987', 'property_reference' => 'abc123', 'short_address' => '1 some road')
+      end
+    end
+
+    context 'when the api root is missing' do
```

Comment:
Should there also be a test that ensures that `api_root` is a valid URL?

Currently a typo such as `htp://hackney.api:8080/` wouldn't raise an error until a request is made against the connection.

Reply:
Good shout - will do

Reply:
As discussed - I don't think there's a way to find errors in the URI in general, since URI.parse will accept most strings.